### ISSUE TITLE
Add more details to CONTRIBUTING.md about project setup?

### ISSUE THREAD

**Issue by NateEag** [streetcomplete/StreetComplete#3505]
**Use case**

As a developer who loves the StreetComplete app, but who hasn't used Java seriously since 2007 (and has never learned Kotlin), I decided to check out the codebase and try building it as a first step towards contributing.

When I first fired up Android Studio, opened the project, and tried a build, I got an error that took me a few minutes to resolve. It said, roughly, that it couldn't build because I hadn't accepted the license agreement for some of the installed tools. I eventually found my way into the Android SDK Manager and got it to let me accept licenses.

Having resolved that, I'm now working on creating a simulated device to deploy to, as the "Play" button failed with the complaint that there were no available targets.

**Proposed Solution**

Would you accept an addition to `CONTRIBUTING.md`'s [Developing section](https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md#development) with details on initial setup of Android Studio?

I realize these might be obvious to an experienced Android developer, and thus not worth documenting explicitly.

I just thought it would be worth noting I hit some friction on initial checkout and asking if it would be worth taking the time to document how I solved that friction.

**Comment by FloEdelmann** [comment]
Just a concern: These instructions need to stay updated to reflect the latest Android Studio changes. Maintaining setup docs is always difficult, as everybody usually just goes through them once and then forgets about them.

I think that linking to good (and hopefully maintained up-to-date) docs is better than maintaining them ourselves. Outdated docs are sonetimes worse than no docs at all.

The setup process for StreetComplete should be quite standard in the Android ecosystem, so there likely is some good documentation / setup tutorial out there.

**Comment by NateEag** [comment]
Yup, I was just coming to the same thoughts myself, @FloEdelmann - there must be good baseline instructions out there we could link to.

I'll see what I can find. Thanks!

**Comment by matkoniecz** [comment]
Welcome! I remember that initial setup was really frustrating, I hope that you will overcome and contribute.

If someone would find good, maintained description of Android Studio pitfalls it would be worth linking. Or maybe creating it somewhere if it does not exist yet.

But it is likely not the best place to document all variants of what may go wrong with basic Android development setup, not specific to this project.

I remember this license agreement mess, but it has not yet made way even into my Ansible setup for Android Studio ( https://github.com/matkoniecz/recovery-with-ansible/blob/master/roles/android-studio/tasks/main.yml ), it is still on "manual actions to run on system restore".

**Comment by NateEag** [comment]
That all makes sense, @matkoniecz. Thanks for the feedback.

I spent a few minutes looking for a good "Set up Android Studio" doc but nothing stands out.

I did find one link under Google's Android Studio docs that covers the licenses situation reasonably well:

https://developer.android.com/studio/intro/update#download-with-gradle

It might be worth adding a sentence to `CONTRIBUTING.md` that points out that first-time build issue, something like:

> If your first build fails, make sure [you have accepted the required license agreements](https://developer.android.com/studio/intro/update#download-with-gradle).

That should give people a good idea issue I bumped into without ballooning the docs.

Sound reasonable?

**Comment by smichel17** [comment]
In a work project of mine, I found it helpful to have a single `DEPENDENCIES.md` file which:

- Lists dependencies **in alphabetical order** (as headers, so GitHub will make it linkable).
- Lists specific key features of those dependencies that we use.
- Links to the relevant external docs for each

The problems I was trying to solve were:

1. Balancing linking to vs including documentation for dependencies.
2. Balancing comprehensive-ness for beginners vs usefulness for experienced people (too much detail on external tools can obscure the information that is unique to your project).

I found this was a good compromise.
